exporter/otlp-metrics/lib/opentelemetry/exporter/otlp/metrics/metrics_exporter.rb

```diff
@@ -0,0 +1,311 @@
+          rescue StandardError => e
+            OpenTelemetry.handle_error(exception: e, message: 'unexpected error in OTLP::MetricsExporter#encode')
+            nil
+          end
+
+          # metrics_pb has following type of data: :gauge, :sum, :histogram, :exponential_histogram, :summary
+          # current metric sdk only implements instrument: :counter -> :sum, :histogram -> :histogram
+          #
+          # metrics [MetricData]
+          def as_otlp_metrics(metrics) # rubocop:disable Metrics/MethodLength
+            case metrics.instrument_kind
+            when :observable_gauge
+              Opentelemetry::Proto::Metrics::V1::Metric.new(
+                name: metrics.name,
+                description: metrics.description,
+                unit: metrics.unit,
+                gauge: Opentelemetry::Proto::Metrics::V1::Gauge.new(
+                  aggregation_temporality: as_otlp_aggregation_temporality(metrics.aggregation_temporality),
+                  data_points: metrics.data_points.map do |ndp|
+                    number_data_point(ndp)
+                  end
+                )
+              )
+
+            when :counter, :up_down_counter
+              Opentelemetry::Proto::Metrics::V1::Metric.new(
+                name: metrics.name,
+                description: metrics.description,
+                unit: metrics.unit,
+                sum: Opentelemetry::Proto::Metrics::V1::Sum.new(
+                  aggregation_temporality: as_otlp_aggregation_temporality(metrics.aggregation_temporality),
+                  data_points: metrics.data_points.map do |ndp|
+                    number_data_point(ndp)
+                  end
+                )
+              )
+
+            when :histogram
+              Opentelemetry::Proto::Metrics::V1::Metric.new(
+                name: metrics.name,
+                description: metrics.description,
+                unit: metrics.unit,
+                histogram: Opentelemetry::Proto::Metrics::V1::Histogram.new(
+                  aggregation_temporality: as_otlp_aggregation_temporality(metrics.aggregation_temporality),
+                  data_points: metrics.data_points.map do |hdp|
+                    histogram_data_point(hdp)
+                  end
+                )
+              )
+            end
+          end
+
+          def as_otlp_aggregation_temporality(type)
+            case type
+            when :delta then Opentelemetry::Proto::Metrics::V1::AggregationTemporality::AGGREGATION_TEMPORALITY_DELTA
+            when :cumulative then Opentelemetry::Proto::Metrics::V1::AggregationTemporality::AGGREGATION_TEMPORALITY_CUMULATIVE
+            else Opentelemetry::Proto::Metrics::V1::AggregationTemporality::AGGREGATION_TEMPORALITY_UNSPECIFIED
+            end
+          end
+
+          def histogram_data_point(hdp)
+            Opentelemetry::Proto::Metrics::V1::HistogramDataPoint.new(
+              attributes: hdp.attributes.map { |k, v| as_otlp_key_value(k, v) },
```

Comment:
Thanks for pointing it out. I was wondering why counter wouldn't have this issue, it turns out the [default value is empty hash](https://github.com/open-telemetry/opentelemetry-ruby/blob/main/metrics_sdk/lib/opentelemetry/sdk/metrics/instrument/counter.rb#L27). 

I would prefer your first suggestion. Do I need to update metrics_sdk here? Will it create another release for metrics_sdk?

Reply:
Ah interesting. I think a separate PR would be a good idea for clarity, let's just make sure we release both the exporter and the SDK at the same time to avoid the bug.

Making a change to metrics_sdk in this PR wouldn't immediately create a release for it. When we run the "Open release request" action it will prepare releases for all gems with unreleased changes labeled with `feat`, `fix`, or `docs`. We can also list specific gems to release if we don't want to release everything.

Reply:
I opened https://github.com/open-telemetry/opentelemetry-ruby/issues/1681 to track this work.